pandas/core/apply.py

```diff
@@ -1427,14 +1427,17 @@ def _managle_lambda_list(aggfuncs: Sequence[Any]) -> Sequence[Any]:
     if len(aggfuncs) <= 1:
         # don't mangle for .agg([lambda x: .])
         return aggfuncs
-    i = 0
+
     mangled_aggfuncs = []
-    for aggfunc in aggfuncs:
-        if com.get_callable_name(aggfunc) == "<lambda>":
-            aggfunc = partial(aggfunc)
-            aggfunc.__name__ = f"<lambda_{i}>"
-            i += 1
-        mangled_aggfuncs.append(aggfunc)
+    for idx, aggfunc in enumerate(aggfuncs):
+        if not com.get_callable_name(aggfunc) == "<lambda>":
+            mangled_aggfuncs.append(aggfunc)
+            continue
+
+        partial_aggfunc = partial(aggfunc)
+        # "partial[Any]" has no attribute "__name__"
+        partial_aggfunc.__name__ = f"<lambda_{idx}>"  # type: ignore[attr-defined]
+        mangled_aggfuncs.append(partial_aggfunc)
```

Comment:
this isn't the same logic - before, `i` was only incremented if `com.get_callable_name(aggfunc) == "<lambda>"`, whereas now you're incrementing `idx` on every iteration